### PR TITLE
app/peerinfo: implement protocol skeleton

### PR DESCRIPTION
**Pull request by corverroos**
This adds the skeleton for the new `peerinfo` protocol. It is aimed at sharing peer information/metadata. 

The supported use-cases are:
 - Track peer clock offset
 - Track peer lock hash
 - Track peer app version 

Tests will be added in a separate PR.

category: feature
ticket: #1224

### CODE REVIEW

File: app/peerinfo/peerinfo.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+)
+
+const period = time.Minute
+
+var protocolID protocol.ID = "/charon/peerinfo/1.0.0"
+
+type (
+	tickerProvider func() (<-chan time.Time, func())
+	nowFunc        func() time.Time
+)
```
Comment (dB2510) [inline app/peerinfo/peerinfo.go:45]: why these types?
for tests?

File: app/peerinfo/peerinfo.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+func NewForT(_ *testing.T, tcpNode host.Host, peers []peer.ID, version string, lockHash []byte,
+	sendFunc p2p.SendReceiveFunc, registerHandler p2p.RegisterHandlerFunc,
+	tickerProvider tickerProvider, nowFunc nowFunc,
+) *PeerInfo {
+	return newInternal(tcpNode, peers, version, lockHash, sendFunc, registerHandler,
+		tickerProvider, nowFunc)
+}
+
+func newInternal(tcpNode host.Host, peers []peer.ID, version string, lockHash []byte,
```
Comment (dB2510) [inline app/peerinfo/peerinfo.go:67]: new internal what?
this function name seems vague

Comment (dB2510) [reply app/peerinfo/peerinfo.go:67]: also add godoc

**Review by dB2510 — APPROVED** [review]

File: app/peerinfo/peerinfo.go (hunk 1 above)
Comment (corverroos) [reply app/peerinfo/peerinfo.go:45]: yup

File: app/peerinfo/peerinfo.go (hunk 2 above)
Comment (corverroos) [reply app/peerinfo/peerinfo.go:67]: `New` and `NewForT` both call `newInternal`

**Pull request merged**